services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/mounted_fs.py

```diff
@@ -87,17 +89,25 @@ def _ensure_directories(self) -> None:
         self.disk_outputs_path  # pylint:disable= pointless-statement
         set(self.disk_state_paths())
 
-    def get_inputs_docker_volume(self) -> str:
-        return f"{self.volume_name_inputs}:{self.inputs_path}"
+    @staticmethod
+    async def _get_bind_path_from_label(label: str) -> Path:
+        volume_details = await get_volume_by_label(label=label)
+        return Path(volume_details["Mountpoint"])
+
+    async def get_inputs_docker_volume(self) -> str:
+        bind_path: Path = await self._get_bind_path_from_label(self.volume_name_inputs)
+        return f"{bind_path}:{self.inputs_path}"
 
-    def get_outputs_docker_volume(self) -> str:
-        return f"{self.volume_name_outputs}:{self.outputs_path}"
+    async def get_outputs_docker_volume(self) -> str:
+        bind_path: Path = await self._get_bind_path_from_label(self.volume_name_outputs)
+        return f"{bind_path}:{self.outputs_path}"
 
-    def get_state_paths_docker_volumes(self) -> Generator[str, None, None]:
+    async def get_state_paths_docker_volumes(self) -> AsyncGenerator[str, None]:
         for volume_state_path, state_path in zip(
             self.volume_name_state_paths(), self.state_paths
         ):
-            yield f"{volume_state_path}:{state_path}"
+            bind_path: Path = await self._get_bind_path_from_label(volume_state_path)
+            yield f"{bind_path}:{state_path}"
 
 
 def setup_mounted_fs() -> MountedVolumes:
```

Comment:
There are two conventions that are violated here:
- ``_mounted_volumes``: we discourage using global variables. Use ``app.state`` instead Read [Data Sharing aka No Singleton](https://docs.aiohttp.org/en/stable/web_advanced.html#data-sharing-aka-no-singletons-please)
- ``get_settings()``: settings should be also in ``app.state`` and init at the same time as the app but a) is yet another singleton b) is created everytime you call this function (ie. does not "get" but "create" )

Reply:
I've already checked and can confirm the the `_mounted_volumes` can be moved to the app instance. For `get_settings()` I am not sure. Will also check.

Reply:
So after spending too much time on trying to move part of the `get_settings()` away. Will stop for now and leave it for a later PR. It's not that straight forward.